Spanner/src/Session/Session.php

```diff
@@ -71,23 +61,28 @@ public function __construct(
         $name
     ) {
         $this->connection = $connection;
-        $this->projectId = $projectId;
-        $this->instance = $instance;
-        $this->database = $database;
-        $this->name = $name;
+        $this->databaseName = SpannerClient::databaseName(
+            $projectId,
+            $instance,
+            $database
+        );
+        $this->name = SpannerClient::sessionName(
+            $projectId,
+            $instance,
+            $database,
+            $name
+        );
     }
 
     /**
      * Return info on the session.
      *
-     * @return array An array containing the `projectId`, `instance`, `database` and session `name` keys.
+     * @return array An array containing the `databaseName` and session `name` keys.
      */
     public function info()
     {
         return [
-            'projectId' => $this->projectId,
-            'instance' => $this->instance,
-            'database' => $this->database,
+            'databaseName' => $this->databaseName,
```

Comment:
Let's keep the old values. This gets a bit close to a backwards-compatibility break. Add the private properties back and set them in the constructor so we don't need to parse the pieces every time this is called.

To be clear: add `private $databaseName` and keep the existing values.